core/src/main/java/org/apache/carbondata/core/datastore/page/statistics/ColumnPageStatsVO.java

```diff
@@ -96,8 +96,7 @@ public void update(Object value) {
       case DOUBLE:
         max = ((double) max > (double) value) ? max : value;
         min = ((double) min < (double) value) ? min : value;
-        int num = getDecimalCount((double) value);
-        decimal = decimal > num ? decimal : num;
+        decimal = getDecimalCount((double) value);
```

Comment:
This seems not correct, it should be updated only if decimal place of this value is longer